spring-boot/src/main/java/org/springframework/boot/jta/narayana/NarayanaRecoveryManagerBean.java

```diff
@@ -53,20 +52,17 @@ public void destroy() throws Exception {
 		this.recoveryManagerService.destroy();
 	}
 
-	void registerXAResourceRecoveryHelper(
-			XAResourceRecoveryHelper xaResourceRecoveryHelper) {
-		getXARecoveryModule(RecoveryManager.manager())
-				.addXAResourceRecoveryHelper(xaResourceRecoveryHelper);
+	void registerXAResourceRecoveryHelper(XAResourceRecoveryHelper xaResourceRecoveryHelper) {
+		getXARecoveryModule().addXAResourceRecoveryHelper(xaResourceRecoveryHelper);
 	}
 
-	private XARecoveryModule getXARecoveryModule(RecoveryManager recoveryManager) {
-		for (RecoveryModule recoveryModule : recoveryManager.getModules()) {
-			if (recoveryModule instanceof XARecoveryModule) {
-				return (XARecoveryModule) recoveryModule;
-			}
+	private XARecoveryModule getXARecoveryModule() {
+		XARecoveryModule xaRecoveryModule = XARecoveryModule.getRegisteredXARecoveryModule();
```

Comment:
It is just a getter which doesn't change any state. Internal logic is the same as the one I'm replacing here.